### PR TITLE
Fixes #7148: Upgrade CFEngine to 3.7.0

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/7148

### CODE REVIEW

File: rudder-agent/SOURCES/Makefile [hunk 1]
```diff
@@ -193,9 +194,6 @@ endif
 	# Apply patches
 	for PATCHNAME in patches/cfengine/*.patch; do echo "Applying $$PATCHNAME..."; $(PATCH) -d ./cfengine-source -p1 < $$PATCHNAME; echo ""; done
 
-	# Bootstrap the package using autogen before compilation
-	cd cfengine-source && NO_CONFIGURE=1 ./autogen.sh
```
Comment (amousset) [inline rudder-agent/SOURCES/Makefile]: Already done in the build-cfengine target (and there is no autogen.sh in the 3.7 tar)

File: rudder-agent/SOURCES/patches/cfengine/02-sighup-support-in-cfserverd.patch [hunk 2]
```diff
@@ -1,82 +0,0 @@
...
-+++ b/cf-serverd/cf-serverd-functions.c	2014-11-28 15:10:21.624969933 +0100
-@@ -356,6 +356,14 @@
-     Log(LOG_LEVEL_DEBUG, "Checking file updates for input file '%s'",
-         config->input_file);
- 
-+    /* If we force the reload of config, effectively recheck the promises */
-+    if ( IsRequestReloadConfig() )
-+    {
-+        Log(LOG_LEVEL_VERBOSE, "Force reload of inputs files...");
-+        GenericAgentCheckPolicy(config, true, true);
```
Comment (amousset) [inline rudder-agent/SOURCES/patches/cfengine/02-sighup-support-in-cfserverd.patch]: Unsure about this one. Config reload on SIGHUP was added but it is done differently, and this [comment](https://github.com/cfengine/core/blob/3.7.0/cf-serverd/cf-serverd-functions.c#L400) is not reassuring.

File: rudder-agent/SOURCES/patches/cfengine/0017-define-classes-when-package-is-up-to-date.patch [hunk 3]
```diff
@@ -1,13 +0,0 @@
...
-@@ -1685,7 +1685,8 @@
-             }
-             else
-             {
--                Log(LOG_LEVEL_VERBOSE, "Installed package is up to date, not updating");
-+                cfPS_HELPER_1ARG(ctx, LOG_LEVEL_VERBOSE, PROMISE_RESULT_NOOP, pp, a,
-+                    "Installed packaged '%s' is up to date, not updating", pp->promiser);
-                 break;
-             }
-         }
```
Comment (amousset) [inline rudder-agent/SOURCES/patches/cfengine/0017-define-classes-when-package-is-up-to-date.patch]: Remove duplicate patch (merge issue).

File: rudder-agent/SOURCES/patches/cfengine/02-sighup-support-in-cfserverd.patch (hunk 2 above)
Comment (amousset) [inline rudder-agent/SOURCES/patches/cfengine/02-sighup-support-in-cfserverd.patch]: The original PR is https://github.com/cfengine/core/pull/2008, the code finally introducing the change is https://github.com/cfengine/core/commit/d0547c25247a7a9fec07585e2e2f93a27b98b8cb. It uses the same code as auto-reload, so I guess It should work as expected.

File: rudder-agent/SOURCES/patches/cfengine/04-close-connection-after-timeout.patch [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
-                 Log(LOG_LEVEL_ERR,
-                     "Timeout - remote end did not respond with the expected amount of data (received=%d, expecting=%d). (recv: %s)",
-                     already, toget, GetErrorStr());
+                 if (LastRecvTimedOut())
+                 {
++                    close(sd);
+                     Log(LOG_LEVEL_ERR,
+                         "Timeout - remote end did not respond with the expected amount of data "
+                         "(received=%d, expecting=%d). (recv: %s)",
+
```
Comment (amousset) [inline rudder-agent/SOURCES/patches/cfengine/04-close-connection-after-timeout.patch]: No functional change, only code formating (cf. https://github.com/cfengine/core/commit/4b2d03bfaeb36fe2a47495f5822f3004b745d6bf)

**Comment by ncharles** [comment]
Awesome, thank you Alexis !

**Pull request merged**